pages/index.js

```diff
@@ -13,10 +15,10 @@ export const getServerSideProps = withIronSessionSsr(
   async function getServerSideProps({ req, res }) {
 
     try {
-      const response = await slsFetch(
-        `${process.env.FRONTEND_URL}/api/settings/mainlogo`
-      );
-      var { data } = await response.json();
+      var rawData = await Setting.get({ slug: mainlogo });    
+      var data = { ...rawData[0], link:`${process.env.KS_S3_BASE_URL}/${rawData[0].value}` };
+  // To ensure maximum efficiency and avoid any unnecessary involvement with the assertUserCan() function, 
```

Comment:
Actually, I'd prefer to use `assertUserCan` inside `getServerSideProps` for maximum security.